docs: prepare for IQP migration (#259)
Qiskit/qiskit-fermions: docs/index.rst

Qiskit Fermions
###############

``qiskit-fermions`` extends the Qiskit SDK with tools for working on fermionic
systems. Within its scope it provides:

- efficient data structures for the representation and manipulation of fermionic
  operators in different forms
- a framework for implementing operator conversion methods (including
  fermion-to-qubit encodings)
- a library of efficient implementations of common conversion methods
- a framework and library of gates for expressing fermionic circuits

> - a transpilation pipeline integrated with Qiskit's transpiler process to

Reviewer: ```suggestion
- a transpilation pipeline integrated with the Qiskit transpiler process to
```

  synthesize fermionic circuits into qubit-based circuits

Additionally, ``qiskit-fermions`` integrates with other tools of the ecosystem,
such as:

- `ffsim <https://qiskit-community.github.io/ffsim/>`_ for efficient simulation
  of fermionic circuits

Getting started
---------------

Several guides exist to help you get started with this package. A good starting
point to get an overview of its breadth of features is the
:doc:`1D Fermi-Hubbard guide <guides/flowsets>`.

Use case examples
-----------------

Components of this package have been used in research related to the following
papers:

- The qDRIFT randomized circuit compilation in [1]_.
- The fermion-to-qubit synthesis during the transpilation process in [2]_.

Technical discussion
--------------------

> Design Intentions

Reviewer: ```suggestion
Design intentions
```

"""""""""""""""""

This package is deliberately designed to align with Qiskit: it builds on a core
implemented in Rust and provides first-party language bindings to Python and C.
Its API intends to draw parallels to Qiskit in order to seamlessly integrate
into the workflows of users with experience in programming Qiskit.

> A core principal to the design of ``qiskit-fermions`` was the decoupling of its

Reviewer: ```suggestion
A core principle to the design of ``qiskit-fermions`` was the decoupling of its
```

fermionic circuit representation from its qubitized form. To be more precise:
the fermionic circuits are meaningful by themselves and do not require a mapping
to qubit space to be interpretable.

> Furthermore, the fermionic circuit representation may not make any assumptions

Author: ```suggestion
Furthermore, the fermionic circuit representation cannot make any assumptions
```

about its fermion-to-qubit encoding applied later on. Consequently, while
Jordan-Wigner retains a dominant position and role, it is not assumed to be the
_default_ fermion-to-qubit encoding.

Known issues
""""""""""""

> As long as Qiskit's own C API has not yet reached feature parity with its Python

Reviewer: ```suggestion
As long as the Qiskit C API has not yet reached feature parity with its Python
```

keep old interfaces and mark them as deprecated, as long as they can co-exist
with the new ones. Each substantial improvement, breaking change, or deprecation
will be documented in the `release notes
<https://quantum.cloud.ibm.com/docs/en/api/qiskit-fermions/release-notes>`_.

References
----------

.. [1] Samuele Piccinelli, et al., `Quantum chemistry with provable convergence
   via randomized sample-based Krylov quantum diagonalization
   <https://arxiv.org/abs/2508.02578v2>`_, arXiv:2508.02578 [quant-ph].

.. [2] Anthony Gandon, et al., `Stabilizer-based quantum simulation of fermion
   dynamics with local qubit encodings <https://arxiv.org/abs/2512.11418v2>`_,
   arXiv:2512.11418 [quant-ph].

.. toctree::
  :hidden:

   Documentation Home <self>
   Installation Instructions <install>
   Guides <guides/index>
   GitHub <https://github.com/Qiskit/qiskit-fermions>

.. toctree::
   :hidden:
   :caption: Tutorials

   Simulate 1D Fermi-Hubbard dynamics with flow sets <guides/flowsets>
   Simulate 2D Fermi-Hubbard dynamics with flow sets <guides/flowsets_2d>

.. toctree::
   :hidden:
   :caption: API reference

>    Python API reference <https://quantum.cloud.ibm.com/docs/en/api/qiskit-fermions>

Author: ```suggestion
   Python API reference <https://quantum.cloud.ibm.com/docs/api/qiskit-fermions>
```

>    C API reference <FIXME>

Author: I was not sure what the C API link has to become

Author: ```suggestion
   C API reference <https://quantum.cloud.ibm.com/docs/api/qiskit-fermions-c>
```
Given some feedback offline, I assume it's this :thinking:
